### PR TITLE
Fix leaking vec in DropDuplicateRows

### PR DESCRIPTION
**Pull request by Pscheidl**
Only reproducible on multinode.

### CODE REVIEW

File: h2o-core/src/main/java/water/rapids/ast/prims/filters/dropduplicates/DropDuplicateRows.java [hunk 1]
```diff
@@ -48,7 +47,7 @@ public Frame dropDuplicates() {
               .outputFrame(Key.make(), sourceFrame.names(), sourceFrame.domains())); // Removing duplicates, domains remain the same
 
       outputFrame = Merge.sort(deDuplicatedFrame, deDuplicatedFrame.numCols() - 1);
-      outputFrame.remove(outputFrame.numCols() - 1);
+      Scope.track(outputFrame.remove(outputFrame.numCols() - 1));
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/rapids/ast/prims/filters/dropduplicates/DropDuplicateRows.java:50]: Obvious leak. The vec gets removed from the frame and scope only tracks the frame. Our scopes do not add all the keys inside once Scope.track is invoked, they only add the top Keyed element and then, cascade is called just in time of deletion.

File: h2o-core/src/test/java/water/rapids/ast/prims/filters/dropduplicates/AstDropDuplicatesTest.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 import static org.junit.Assert.*;
 
 @RunWith(H2ORunner.class)
-@CloudSize(1)
+@CloudSize(5)
```
Comment (honzasterba) [inline h2o-core/src/test/java/water/rapids/ast/prims/filters/dropduplicates/AstDropDuplicatesTest.java:20]: probably unintentional

Comment (Pscheidl) [reply h2o-core/src/test/java/water/rapids/ast/prims/filters/dropduplicates/AstDropDuplicatesTest.java:20]: Yes. Thanks.

**Comment by Pscheidl** [comment]
The CRAN check looks ok:

```
* using log directory â€˜/home/jenkins/slave_dir_from_mr-0xc1/workspace/h2o-3-pipeline_PR-4657/r3.5-cmd-check-as-cran/h2o-3/h2o-r/R/src/contrib/h2o.Rcheckâ€™
* using R version 3.5.3 (2019-03-11)
* using platform: x86_64-pc-linux-gnu (64-bit)
* using session charset: UTF-8
* using option â€˜--as-cranâ€™
* checking for file â€˜h2o/DESCRIPTIONâ€™ ... OK
* checking extension type ... Package
* this is package â€˜h2oâ€™ version â€˜3.30.0.99999â€™
* checking CRAN incoming feasibility ... NOTE
Maintainer: â€˜Erin LeDell <erin@h2o.ai>â€™

Version contains large components (3.30.0.99999)

Package has FOSS license, installs .class/.jar but has no 'java' directory.

Size of tarball: 128356488 bytes
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking serialization versions ... OK
* checking whether package â€˜h2oâ€™ can be installed ... OK
* checking installed package size ... NOTE
  installed size is 132.5Mb
  sub-directories of 1Mb or more:
    R       1.0Mb
    java  130.1Mb
* checking package directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking use of S3 registration ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... NOTE
Package unavailable to check Rd xrefs: â€˜plyrâ€™
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking examples ... OK
* checking PDF version of manual ... OK
* DONE
Status: 3 NOTEs
```

**Comment by honzasterba** [comment]
R check failed because of java.io.IOException: Unexpected termination of the channel

CRAN check failed because

[2020-05-28T18:28:59.466Z] There is 1 unexpected NOTE in R CMD check --as-cran
[2020-05-28T18:28:59.466Z] * checking installed package size
[2020-05-28T18:28:59.466Z]     R       1.0Mb

**Comment by Pscheidl** [comment]
Jenkins went down completely and we lost all test results. Now it seems to be up again. Re-running.

**Comment by Pscheidl** [comment]
http://mr-0xc1:8080/job/h2o-3-pipeline/job/PR-4657/2/

**Comment by Pscheidl** [comment]
Build trigger did not work, therefore I started a new build manually: http://mr-0xc1:8080/job/h2o-3-pipeline/job/PR-4657/4/

**Comment by Pscheidl** [comment]
Aborted by Anonymous, ran once again.

**Pull request merged**